src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -54,6 +54,10 @@
 #define GRPC_OBJC_VERSION_STRING @"1.0.0"
 
 static NSMutableDictionary *kHostCache;
+
+// This connectivity monitor flushes the host cache when connectivity status
+// changes. Otherwise, a new call might still use the cached channel which is
+// no longer available and will call gRPC to hang
```

Comment:
Well, the channel is available, and WiFi coming up doesn't mean we've lost cell connectivity!